### PR TITLE
A solution to #12

### LINKED ISSUE #12: Message.channel is a DMChannel in on_message for an ephemeral message not sent as a DM
### Link

https://github.com/Rapptz/discord.py/issues/7370

### PR DESCRIPTION
**Pull request by veni-vidi-code**
## Summary

Making channel a property for messages to resolve ephermal messages correctly
Closes #12 

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/message.py [hunk 1]
```diff
@@ -962,6 +963,13 @@ def repl(obj):
         result = pattern.sub(repl, self.content)
         return escape_mentions(result)
 
+    @utils.cached_slot_property('_cs_channel')
+    def channel(self) -> MessageableChannel:
+        if self.flags.ephemeral and isinstance(self._channel, DMChannel):
+            self._channel = self._state.get_channel(self._channel.id)
+            self.guild = self._channel.guild
```
Comment (TAG-Epic) [inline nextcord/message.py:970]: guild would now only be accessible if you used .channel first?

Comment (veni-vidi-code) [reply nextcord/message.py:970]: i will change this

**Comment by ooliver1** [comment]
This is pretty old and there may be a better way to fix this, I'm gonna close it for now

**Comment by ooliver1** [comment]
> This is pretty old and there may be a better way to fix this, I'm gonna close it for now

**Comment by ooliver1** [comment]
mb

**Pull request closed without merging**